### PR TITLE
Add links and apply link encoding in Middleware doc

### PR DESCRIPTION
**Pull request by scottaddie**
Encode asterisks in existing links and add a few new links.

### CODE REVIEW

File: aspnetcore/fundamentals/middleware/index.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ description: Learn about ASP.NET Core middleware and the request pipeline.
 monikerRange: '>= aspnetcore-2.1'
```
Comment (wadepickett) [inline aspnetcore/fundamentals/middleware/index.md:5]: Low pri nit suggestion: Consider "exit" as opposed to "short-circuit"

File: aspnetcore/fundamentals/middleware/index.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ Middleware is software that's assembled into an app pipeline to handle requests
...
+Request delegates are configured using <xref:Microsoft.AspNetCore.Builder.RunExtensions.Run%2A>, <xref:Microsoft.AspNetCore.Builder.MapExtensions.Map%2A>, and <xref:Microsoft.AspNetCore.Builder.UseExtensions.Use%2A> extension methods. An individual request delegate can be specified in-line as an anonymous method (called in-line middleware), or it can be defined in a reusable class. These reusable classes and in-line anonymous methods are *middleware*, also called *middleware components*. Each middleware component in the request pipeline is responsible for invoking the next component in the pipeline or short-circuiting the pipeline. When a middleware short-circuits, it's called a *terminal middleware* because it prevents further middleware from processing the request.
```
Comment (wadepickett) [inline aspnetcore/fundamentals/middleware/index.md:25]: Low pri nit suggestion: Consider "exit" as opposed to "short-circuit"

File: aspnetcore/fundamentals/middleware/index.md [hunk 3]
```diff
@@ -298,9 +298,9 @@ The simplest possible ASP.NET Core app sets up a single request delegate that ha
 
 [!code-csharp[](index/snapshot/Middleware/Startup.cs)]
 
-The first <xref:Microsoft.AspNetCore.Builder.RunExtensions.Run*> delegate terminates the pipeline.
+The first <xref:Microsoft.AspNetCore.Builder.RunExtensions.Run%2A> delegate terminates the pipeline.
```
Comment (wadepickett) [inline aspnetcore/fundamentals/middleware/index.md:301]: Consider: In the previous section, original to the doc:  The introduction of a single request delegate pattern using a single anonymous function might benefit from a sub title for it.  Then the reader can go right to that pattern if that is what they are looking for.  Perhaps the "Chain multiple request delegates together with Use" section could also use a sub title for the same reason.

Comment (scottaddie) [reply aspnetcore/fundamentals/middleware/index.md:301]: Thanks for the suggestion. I'll open another PR to address that.

**Pull request merged**